selfdrive/common/messagehelp.h

```diff
@@ -40,3 +40,56 @@ class MessageReader {
   capnp::FlatArrayMessageReader *msg_reader;
   cereal::Event::Reader event;
 };
+
+#define SEGEMENT_BUF_SIZE 256
+class MessageBuilder : public capnp::MessageBuilder {
+ public:
+  MessageBuilder() : returnedFirstSegment(false), nextSize(2048), stackSegment{} {};
+
+  ~MessageBuilder() {
+    for (auto ptr : moreSegments) {
+      free(ptr);
+    }
+  }
+
+  kj::ArrayPtr<capnp::word> allocateSegment(uint minimumSize) {
+    if (!returnedFirstSegment) {
+      returnedFirstSegment = true;
+      uint size = kj::max(minimumSize, SEGEMENT_BUF_SIZE - 1);
+      if (size < SEGEMENT_BUF_SIZE) {
+        return kj::ArrayPtr<capnp::word>(stackSegment + 1, size);
+      }
+    }
+    uint size = kj::max(minimumSize, nextSize);
+    capnp::word *result = (capnp::word *)calloc(size, sizeof(capnp::word));
+    moreSegments.add(result);
+    nextSize += size;
+    return kj::ArrayPtr<capnp::word>(result, size);
+  }
+
+  kj::ArrayPtr<capnp::word> toFlatArrayPtr() {
+    kj::ArrayPtr<const kj::ArrayPtr<const capnp::word>> segments = getSegmentsForOutput();
```

Comment:
Hmm, does "word>>" compile? I think it used to have to be "word> >"

Reply:
>> is ok. an "auto"  maybe more clearly.